guide-en/README.md

```diff
@@ -159,5 +148,288 @@ exports.default = commandModule({
         }
 });
 ```
+Commands are straight forward. Keep in mind, every other property on the commandModule object is 
+optional **except** the type and execute function.
+
+# Context class
+The provided Context class helps with modules of `CommandType.Both` ( A mixture of slash / legacy commands )
+In short, Slash Commands, Both Commands, Text Commands carry a Context data structure
 
+The Context class is passed into modules with type :
+- `CommandType.Both`
+- `CommandType.Slash`
+- `CommandType.Text`
+
+Typescript:
+```typescript
+export default commandModule({
+        name: 'ping',
+        type: CommandType.Both,
+        async execute(ctx: Context) {
+            await ctx.reply(`pong ${ctx.user}`) 
+            // .reply is shared between both message and interaction!
+            // So is an User object!
+        }
+});
+```
+Javascript:
+```javascript
+exports.default = commandModule({
+        name: 'ping',
+        type: CommandType.Both,
+        async execute(ctx) { //ctx is a Context instance
+            await ctx.reply(`pong ${ctx.user}`) 
+            // .reply is shared between both message and interaction!
+            // So is an User object!
+        }
+});
+```
+
+# Plugins
+
+As of now, modules seem a little underwhelming. It appears that sern doesn't have all the features of a standard handler, 
+which manages permissions, categorizes, cool-downs, publishes application commands, role permissions, etc. Many important 
+parts that manage access and help streamline command creation to make are apparently absent.  
+Below is an example of an event plugin, one of the types of plugins.  
+
+Typescript:
+```typescript
+export function serenOnly(): EventPlugin<CommandType.Both> {
+  return {
+    type: PluginType.Event,
+    async execute([ctx, args], controller) {
+      if (ctx.user.id !== "182326315813306368") {
+        await ctx.reply({content: "You cannot use this command"})
+        return controller.stop()
+      }
+      return controller.next();
+    }
+  }
+}
+```
+Javascript:
+```javascript
+export function serenOnly(): EventPlugin<CommandType.Both> {
```

Comment:
seriously? dude this is javascript

Reply:
Lmao

Reply:
Oops lol I copied typescript example over, forgot to remove types

Reply:
fixed